components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/DataFlowProcessingEngine.cs

```diff
@@ -62,6 +62,7 @@ public class DataFlowProcessingEngine : IProcessingEngine, IDisposable {
         /// <inheritdoc/>
         public void Dispose() {
             _messageTrigger.OnMessage -= MessageTriggerMessageReceived;
+            _messageTrigger.Dispose();
```

Comment:
Autofac should be calling it.  Otherwise must override onrelease during registration or better, consume it as Owned<>

Reply:
done

Reply:
I did not suggest this was not valid - e.g. if you want to control the order of dispose call then it makes sense to take ownership of the instance passed to you.

Reply:
actually this helped on a different bug that was still open. Actually the original problem was that the Message trigger was not implementing the IDisposable...